ts-client/src/pool.ts

```diff
@@ -39,59 +46,55 @@ function fromTokenMintsToCompositeKey(
   return new PublicKey(compositeKeyBuffer);
 }
 
-class Pool {
-  private program: AmmProgram;
-
-  private vaultA: Vault;
-  private vaultB: Vault;
-
-  private vaultASpl: VaultSpl;
-  private vaultBSpl: VaultSpl;
-
-  private poolSpl: PoolSpl;
-  private onChainTime: number = 0;
-
-  public state: PoolState | undefined;
-  private swapCurve: SwapCurve | undefined;
+export class Pool {
+  private swapCurve: SwapCurve;
+
+  constructor(
+    private program: AmmProgram,
+    private vaultA: Vault,
+    private vaultB: Vault,
+    public state: PoolState,
+    private vaultASpl: VaultSpl,
+    private vaultBSpl: VaultSpl,
+    private poolSpl: PoolSpl,
+    private onChainTime: number = 0
+  ) {
+    if ("stable" in this.state.curveType) {
+      this.swapCurve = new StableSwap(
+        this.state.curveType.stable.amp.toNumber()
+      );
+    } else {
+      this.swapCurve = new ConstantProductSwap();
+    }
```

Comment:
Can we convert this to getter?

```
get swapCurve() {
      invariant(this.state.curveType, 'Curve Type not found')
      return "stable" in this.state.curveType ? new StableSwap(this.state.curveType.stable.amp.toNumber()) : new ConstantProductSwap();
}

Reply:
It is better to throw in the constructor if the created object isn't valid. Is there any advantage in making this fail later?

Reply:
ohh no, just though of to save an extra declaration. I think your point make more sense